client/components/relatedProducts/Card.jsx

```diff
@@ -5,30 +5,74 @@ import { AiOutlineStar } from 'react-icons/ai';
 import StarRating from '../StarRating';
 
 import getStarRatingAvg from '../../utils/getStarRatingAvg';
-import getImageUrl from '../../utils/getImageUrl';
+import getPhotos from '../../utils/getImageUrl';
 
 import '../../css/relatedProducts/Card.scss';
 
-const STAR_SIZE = 24;
+const STAR_SIZE = 16;
 
 const Card = ({
   relatedProduct, handleActionBtnClick, handleCardClick,
 }) => {
   const [rating, setRating] = useState(0);
-  const [imageUrl, setImageUrl] = useState('');
-  const hasNoRating = Number.isNaN(rating);
+  const [photos, setPhotos] = useState([]);
+  const [mainImageUrl, setMainImageUrl] = useState('');
 
   useEffect(() => {
     getStarRatingAvg(relatedProduct.id).then((result) => setRating(result.avg));
-    getImageUrl(relatedProduct.id).then((url) => setImageUrl(url));
+    getPhotos(relatedProduct.id).then((images) => {
+      const firstImage = images[0];
+
+      setMainImageUrl(firstImage.url);
+      setPhotos(images.slice(0, 5));
+    });
   }, []);
 
+  const handleThumbnailClick = (e) => {
+    e.stopPropagation();
+    const { id } = e.target;
+    const photosCopy = photos.slice();
+    const temp = photosCopy[0];
+    photosCopy[0] = photosCopy[id];
+    photosCopy[id] = temp;
+
+    setMainImageUrl(photosCopy[0].url);
+    setPhotos(photosCopy);
+  };
+
+  const handleKeyPress = (e) => {
+    e.stopPropagation();
+    const { id } = e.target;
+    const photosCopy = photos.slice();
+    const temp = photosCopy[0];
+    photosCopy[0] = photosCopy[id];
+    photosCopy[id] = temp;
+
+    setMainImageUrl(photosCopy[0].url);
+    setPhotos(photosCopy);
+  };
+
   return (
-    <div className="related-products__card" role="button" tabIndex={0} onKeyPress={() => { handleCardClick(relatedProduct.id); }} onClick={() => { handleCardClick(relatedProduct.id); }}>
-      <div className="related-products__card-image" style={imageUrl ? { backgroundImage: `url(${imageUrl})` } : {}}>
-        <span className="related-products__action-btn" role="button" tabIndex={0} onKeyPress={() => { handleActionBtnClick(relatedProduct); }} onClick={() => handleActionBtnClick(relatedProduct)}>
-          <AiOutlineStar size={STAR_SIZE} />
+    <div className="related-products__card" role="button" tabIndex={0} onKeyPress={() => { handleCardClick(relatedProduct.id); }} onClick={() => handleCardClick(relatedProduct.id)}>
+      <div className="related-products__card-image" style={mainImageUrl ? { backgroundImage: `url(${mainImageUrl})` } : {}}>
+        <span className="related-products__action-btn" role="button" tabIndex={0} onKeyPress={() => { handleActionBtnClick(relatedProduct); }} onClick={(e) => { e.stopPropagation(); handleActionBtnClick(relatedProduct); }}>
```

Comment:
Put the attributes in a new line might be more readable, and something will break the max 100 characters a line rule.